cmd/tink-worker/internal/registry_test.go

```diff
@@ -0,0 +1,72 @@
+package internal
+
+import (
+	"context"
+	"errors"
+	"io"
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/packethost/pkg/log"
+	"github.com/stretchr/testify/assert"
+)
+
+func setupTestLogger() log.Logger {
+	service := "github.com/tinkerbell/tink"
+	logger, err := log.Init(service)
+	if err != nil {
+		panic(err)
+	}
+	return logger
+}
+
+type imagePullerMock struct{}
+
+func (d *imagePullerMock) ImagePull(ctx context.Context, str string, op types.ImagePullOptions) (io.ReadCloser, error) {
+	var s string
+	var err error
+	if str == "test/success" {
+		s = "{\"status\": \"hello\",\"error\":\"\"}{\"status\":\"world\",\"error\":\"\"}"
+	} else if str == "test/fail" {
+		s = "{\"error\": \"\"}"
+		err = errors.New("Tested, failure of the image pull")
+	} else if str == "test/fail_partial" {
+		s = "{\"status\": \"hello\",\"error\":\"\"}{\"status\":\"world\",\"error\":\"Tested, failure of No space left on device\"}"
```

Comment:
It would be nice to have this data defined directly in the test case, to keep things simple.

Reply:
Initially, I thought it could be better, but then I have made changes as per your suggestion.